courses/fundamentals_of_ada/040_statements.rst

```diff
@@ -655,202 +645,145 @@ Low-Level For-loop Statements
-   Discrete subtype definition - loop parameter type and range of values
-
------------------------------------
-Simple Low-Level For-loop Example
------------------------------------
+* Example
 
 .. code:: Ada
 
-   procedure Demo_LowLevel_ForLoops is
-     type Days is (Sun, Mon, Tue, Wed, Thu, Fri, Sat);
-   begin
-     for Today in Days range Mon .. Fri loop
-       Put_Line( Days'Image(Today) );
-     end loop;
-     for K in Integer range -1 .. 10 loop
-       Put_Line( Integer'Image(K) );
+     for Day in Days_T loop
+        Refresh_Planning (Day);
      end loop;
-   end Demo_LowLevel_ForLoops;
+ 
+.. container:: speakernote
+
+   Name - loop parameter object
+   Discrete subtype definition - loop parameter type and range of values
  
 -----------------------------------
-Expressing the Sequence of Values
+Variable and Sequence of Values
 -----------------------------------
 
-* Several syntactic forms allowed
-* Just write what seems natural, it is likely legal
-
-.. code:: Ada
-
-   -- name without constraint
-   for This_Day in Days loop
-   -- name with constraint
-   for This_Day in Days range Mon .. Fri
-   -- named subtype
-   for This_day in Weekdays loop
-   -- expression .. Expression
-   for This_day in Mon .. Fri loop
-   -- variable(s) of type Days
-   for This_day in Start .. Stop loop
- 
--------------------------------
-Low-Level For-loop Parameters
--------------------------------
+* Variable declared implicitly by loop statement
 
-* Declared implicitly by loop statement
-* Has a constant view
+   - Has a view as constant
+   - No assignment or update possible
 
-   - No assignment
-   - No update via subprogram parameter
+* Initialized as :ada:`'First`, incremented as :ada:`'Succ`
 
-* Step-size is always logically "one"
+* Syntaxic sugar: several forms allowed
 
 .. code:: Ada
-    
-   type Days is (Mon, Tues, Wed, Thurs, Fri, Sat, Sun);
-
-   for Day in Days loop
-      Days_IO.Put (Day);
-      New_Line;
-   end loop;
-   
-   for Day in Days range Mon .. Fri loop
-
-   for Count in Leaf_Counter range 1 .. 10 loop
 
-     
+   -- All values of a type or subtype
+   for Day in Days_T loop
+   for Day in Days_T range Mon .. Fri -- anonymous subtype
+   -- Constant and variable range
+   for Day in Mon .. Fri loop
+   Today, Tomorrow : Days_T;
+   ...
+   for Day in Today .. Tomorrow loop
+ 
 -----------------------------------
 Low-Level For-loop Parameter Type
 -----------------------------------
 
-* Need not be specified explicitly
+.. admonition:: Language Variant
+    
+    Ada 2012
 
-   - As long as can be deduced properly from range
+* The type can be implicit 
 
-   .. code:: Ada
+   - As long as it is clear for the compiler
+   - Warning: same name can belong to several enums
 
-      for Today in Mon .. Fri loop
- 
-* Predefined type `Integer` assumed for convenience
+   .. code:: Ada
 
-   - Each bound must be either a
+      -- Error if Wed is both Days_T and Marital_Status_T
```

Comment:
No - error only if Wed and Fri are BOTH in two types - otherwise compiler can figure it out.
Better example "type Rgb_t is ( red, green, blue ); type stoplight_t is ( red, yellow, green); ... for x in red .. green loop"